wire-ios-data-model/Source/E2EIdentity/E2eIVerificationStatusService.swift

```
@@ -0,0 +1,93 @@
+//
+
+import Foundation
+import WireCoreCrypto
+
+// sourcery: AutoMockable
+public protocol E2eIVerificationStatusServiceInterface {
+
+    func getConversationStatus(groupID: MLSGroupID) async throws -> MLSVerificationStatus
+
+}
+
+public final class E2eIVerificationStatusService: E2eIVerificationStatusServiceInterface {
+
+    // MARK: - Properties
+
+    private let coreCryptoProvider: CoreCryptoProviderProtocol
+    private var coreCrypto: SafeCoreCryptoProtocol {
+        get async throws {
+            try await coreCryptoProvider.coreCrypto(requireMLS: true)
+        }
+    }
+
+    // MARK: - Life cycle
+
+    public init(coreCryptoProvider: CoreCryptoProviderProtocol) {
+        self.coreCryptoProvider = coreCryptoProvider
+    }
+
+    // MARK: - Error
+
+    public enum E2eIVerificationStatusError: Error {
+
+        case missingConversation
+        case failedToFetchVerificationStatus
+
+    }
+
+    // MARK: - Public interface
+
+    /// Returns the state of a conversation regarding end-to-end identity.
+    /// Note: coreCrypto indicates a conversation with one of the states: `verified`, `notVerified`, `notEnabled`.
+    /// For further use, we need to convert it to MLSVerificationStatus, which has `verified`, `notVerified`, and `degraded` states.
+    ///
+    /// - Parameters:
+    ///  - groupID: the id of the MLS group for which to get the verification status
+    /// - Throws:
+    ///   `E2eIVerificationStatusError` if the status couldn't be fetched
+    /// - Returns:
+    /// - `MLSVerificationStatus`
+
+    public func getConversationStatus(groupID: MLSGroupID) async throws -> MLSVerificationStatus {
+        do {
+            return try await coreCrypto.perform {
+                try await $0.e2eiConversationState(conversationId: groupID.data).toMLSVerificationStatus()
+            }
+        } catch {
+            throw E2eIVerificationStatusError.failedToFetchVerificationStatus
```

The actual error is lost at this point and nowhere logged, if I'm not mistaken.

---

https://github.com/wireapp/wire-ios/pull/1040

---

added logs